cmd/dfdaemon/app/init.go

```diff
@@ -0,0 +1,159 @@
+package app
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/dragonflyoss/Dragonfly/common/dflog"
+	dferr "github.com/dragonflyoss/Dragonfly/common/errors"
+	"github.com/dragonflyoss/Dragonfly/common/util"
+	"github.com/dragonflyoss/Dragonfly/dfdaemon/config"
+	"github.com/dragonflyoss/Dragonfly/dfdaemon/constant"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// initDfdaemon sets up running environment for dfdaemon according to the given config
+func initDfdaemon(cfg config.Properties) error {
+	// if Options.MaxProcs <= 0, programs run with GOMAXPROCS set to the number of cores available.
+	if cfg.MaxProcs > 0 {
+		runtime.GOMAXPROCS(cfg.MaxProcs)
+	}
+
+	if err := initLogger(cfg); err != nil {
+		return errors.Wrap(err, "init logger")
+	}
+
+	if cfg.Verbose {
+		logrus.Infoln("use verbose logging")
+	}
+
+	if err := os.MkdirAll(cfg.DFRepo, 0755); err != nil {
+		return dferr.Newf(
+			constant.CodeExitRepoCreateFail,
+			"ensure local repo %s exists", cfg.DFRepo,
+		)
+	}
+
+	go cleanLocalRepo(cfg.DFRepo)
+
+	dfgetVersion, err := exec.Command(cfg.DFPath, "version").CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "get dfget version")
+	}
+	logrus.Infof("use dfget %s from %s", bytes.TrimSpace(dfgetVersion), cfg.DFPath)
+
+	return nil
+}
+
+// initLogger initialize the global logrus logger
+func initLogger(cfg config.Properties) error {
+	current, err := user.Current()
+	if err != nil {
+		return errors.Wrap(err, "get current user")
+	}
+	logFilePath := filepath.Join(current.HomeDir, ".small-dragonfly/logs/dfdaemon.log")
+	logrus.Infof("use log file %s", logFilePath)
```

Comment:
IMOP, it's better to output it on debug level.

Reply:
Sure